src/OrleansRuntime/MembershipService/MembershipOracle.cs

```diff
@@ -38,16 +38,16 @@ internal class MembershipOracle : SystemTarget, IMembershipOracle, IMembershipSe
         public SiloAddress SiloAddress { get { return membershipOracleData.MyAddress; } }
         private TimeSpan AllowedIAmAliveMissPeriod { get { return orleansConfig.Globals.IAmAliveTablePublishTimeout.Multiply(orleansConfig.Globals.NumMissedTableIAmAliveLimit); } }
 
-        public MembershipOracle(Silo silo, MembershipTableFactory membershipTableFactory, IInternalGrainFactory grainFactory)
-            : base(Constants.MembershipOracleId, silo.SiloAddress)
+        public MembershipOracle(ILocalSiloDetails siloDetails, ClusterConfiguration clusterConfiguration, NodeConfiguration nodeConfiguration, MembershipTableFactory membershipTableFactory, IInternalGrainFactory grainFactory)
+            : base(Constants.MembershipOracleId, siloDetails.SiloAddress)
```

Comment:
Intention here is to break the dependency on Silo, since Silo is now being constructed by DI when the builder is being used.